pkgs/by-name/la/labelife-label-printer/package.nix

```diff
@@ -0,0 +1,53 @@
+{ lib, stdenv, fetchurl, cups, autoPatchelfHook }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "labelife-label-printer";
+  version = "1.2.1";
+
+  arch =
+    if stdenv.hostPlatform.system == "x86_64-linux" then "x86_64"
+    else if stdenv.hostPlatform.system == "i686-linux" then "i386"
+      else throw "Unsupported system: ${stdenv.hostPlatform.system}";
+
+  src = fetchurl {
+    url = "https://oss.saas.aimocloud.com/saas/Lablife/bag/LabelPrinter-${finalAttrs.version}.tar.gz";
+    hash = "sha256-twnIFMBMyEM3xGlsuk3763C3emz3mgpEnlfvnL0XRWw=";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [ cups ];
+
+  installPhase =
+  ''
+    # Install the CUPS filter with executable permissions
+    install -Dm755 ./${finalAttrs.arch}/rastertolabeltspl $out/lib/cups/filter/rastertolabeltspl
+
+    # Install all PPD files with read and write permissions for owner, and read for group and others
+    for ppd in ./ppds/*.ppd; do
+      install -Dm644 $ppd $out/share/cups/model/label/$(basename $ppd)
+    done
+  '';
+
+  meta = with lib; {
```

Comment:
tips:

1. Remove `with lib;`,
2. Sort attributes alphabetically.

Reply:
This is a very opinionated nit-pick 😅. I've re-ordered the attributes to match their order of appearance in the [nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#chap-meta) in a00854bc080107ed3d042e9991bfc6f1f3097647, and kept using `with lib;` (DRY) like as shown in the manual.

Reply:
Hey,

This is not mostly not opinionated.

1. There's multiple initiative going on to remove `with lib;`, see it by yourself: https://github.com/NixOS/nixpkgs/pull/293767
2. That might be the most opinionated comment indeed, it's usually better to keep attributes alphabetically in `meta` to ease finding them. For the top attributes, keeping them by appearance is recommended.